ConsoleHabitTracker.kraven88/DataAccess/SqliteDB.cs

```diff
@@ -0,0 +1,258 @@
+    }
+	
+	internal List<DailyProgress> LoadCurrentProgress(Habit habit)
+	{
+        var sql =
+                $@"SELECT *
+				FROM DailyProgress
+				WHERE HabitId = {habit.Id}
+				ORDER BY Date DESC
+				LIMIT 1;";
+
+		return LoadProgressList(sql);
+    }
+    
+	internal List<Habit> LoadExistingHabits()
+    {
+		var habits = new List<Habit>();
+		using (var connection = new SQLiteConnection(connectionString))
+		{
+			connection.Open();
+			var sql = connection.CreateCommand();
+			sql.CommandText = "SELECT * FROM Habits";
+
+			var reader = sql.ExecuteReader();
+			if (reader.HasRows)
+			{
+				while (reader.Read())
+				{
+					var habit = new Habit();
+					habit.Id = reader.GetInt32(0);
+					habit.Name = reader.GetString(1);
+					habit.UnitOfMeasurement = reader.GetString(2);
+					habit.CurrectGoal = reader.GetInt32(3);
+
+					habits.Add(habit);
+				}
+			}
+			else
+				throw new ArgumentException("No habits found.");
+
+			reader.Close();
+			connection.Close();
+		}
+
+		return habits;
+    }
+	
+	internal Habit LoadHabit(string habitName)
+	{
+		using (var connection = new SQLiteConnection(connectionString))
+		{
+			connection.Open();
+			var sql = connection.CreateCommand();
+			var habit = new Habit();
+
+			sql.CommandText =
+				$@"SELECT *
+				FROM Habits
+				WHERE Name=""{habitName}""";
+
+			var reader = sql.ExecuteReader();
+
+			if (reader.HasRows)
+			{
+				while (reader.Read())
+				{
+					habit.Id = reader.GetInt32(0);
+					habit.Name = reader.GetString(1);
+					habit.UnitOfMeasurement = reader.GetString(2);
+					habit.CurrectGoal = reader.GetInt32(3);
+				}
+			}
+			else throw new ArgumentException($"Couldn't find habit with the name: {habitName}");
+
+			reader.Close();
+
+			return habit;
+		}
+	}
+	
+	private List<DailyProgress> LoadProgressList(string sqlCommand)
+	{
+		var progressList = new List<DailyProgress>();
+		using (var connection = new SQLiteConnection(connectionString))
+		{
+			connection.Open();
+			var sql = connection.CreateCommand();
+			sql.CommandText = sqlCommand;
+
+            var reader = sql.ExecuteReader();
+			if (reader.HasRows)
+			{
+				while (reader.Read())
+				{
+					var daily = new DailyProgress();
+					daily.Id = reader.GetInt32(0);
+					daily.Date = DateOnly.ParseExact(reader.GetString(2), "dd.MM.yyyy");
+					daily.Quantity = reader.GetInt32(3);
+					daily.DailyGoal = reader.GetInt32(4);
+
+					progressList.Add(daily);
+				}
+			} else throw new ArgumentException($"Couldn't find Progress List.");
+
+			reader.Close();
+			connection.Close();
+        }
+
+		return progressList;
+	}
+	
+	private void SaveData(string sqlCommand)
+	{
+		using (var connection = new SQLiteConnection(connectionString))
+		{
+			connection.Open();
+			var sql = connection.CreateCommand();
+			sql.CommandText = sqlCommand;
+			sql.ExecuteNonQuery();
+			connection.Close();
+		}
+	}
+    
+	internal void SaveNewHabit(string name, string unit)
+    {
+		var sql = 
```

Comment:
I'm getting an error after saving a habit, I think because there's something missing from your query.

![image](https://user-images.githubusercontent.com/108711476/216592998-48316303-0f2a-4d30-880e-6c2fe46fbd26.png)

![image](https://user-images.githubusercontent.com/108711476/216593085-693492b9-d04b-4fba-aaac-1ee6c35bdab7.png)

Reply:
Oh god, how did I missed that